test/aws/osml/model_runner/tile_worker/test_tiling_strategy.py

```diff
@@ -32,13 +40,21 @@ def test_chip_generator(self):
         assert chip_list[4] == ((2048, 2048), (2048, 452))
         assert chip_list[5] == ((2048, 4096), (904, 452))
 
+        # Test case 3: Image with full overlap, large crop sizes
```

Comment:
I know this PR is focused on comments but this test is really 3 tests and should be written as such.  Instead of `test_chip_generator` with 3 cases defined in comments it should be `test_chip_generator_partial_overlap`, `test_chip_generator_no_overlap`, and `test_chip_generator_full_overlap` with further elaboration (if needed) in the docstrings.  Cleaner code, reduced comment clutter, and test failures point directly to the reason.

Reply:
Broke out the test cases like suggested.